File: Programming-Languages-Concepts-and-Paradigms/Laboratory/04-lab/practice.cpp
#include <iostream>
#include <string>

using namespace std;

int main() {
  string s1 = "This is the house that Jack built.";
  cout << s1.substr(0,4) << endl;
  cout << s1.substr(5,2) << endl;
  cout << s1.find("This") << endl;
  cout << s1.find('J') << endl;
  cout << s1.find("is", 4) << endl;
  cout << s1.length() << endl;
  cout << s1.erase(0,5) << endl;
  cout << s1.length() << endl;
}

File: Programming-Languages-Concepts-and-Paradigms/Laboratory/05-lab/practice.go
package main

import (
	"fmt"
	"strings"
)

func main() {
	s1 := "This is the house that Jack built."

	fmt.Println(s1[:4])
	fmt.Println(s1[5:7])
	fmt.Println(strings.Index(s1, "This"))
	fmt.Println(strings.Index(s1, "J"))
	fmt.Println(strings.Index(s1[4:], "is") + 4)
	fmt.Println(len(s1))
	s1 = s1[5:]
	fmt.Println(s1)
	fmt.Println(len(s1))
}
